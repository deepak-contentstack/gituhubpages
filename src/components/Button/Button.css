@import '../../../tokens/variables.scss';
@import '../../../tokens/tokens.css';
@import '../../static/styles/css/typography.css';
@import '../../static/styles/css/common.css';

[type='button']:not(:disabled),
[type='reset']:not(:disabled),
[type='submit']:not(:disabled),
button:not(:disabled) {
  	cursor: pointer;
}

.ui-component button{
  	margin: 0 calc(var(--space-2) + 1px);
}

/* Button */

.Button {
	background-color: transparent;
	border: 1px solid transparent;
	border-radius: $radii-6;
	font-family: $font-family-primary;
	font-size: $size-font-medium;
	font-weight: $font-weight-bold;
	line-height: $space-12;
	/* line-height: calc(var(--space-12) + 1px); */
	min-height: $space-32;
	min-width: $space-32;
	padding: $space-8 $space-16;
	/* padding: $space-8 $space-16 calc(var(--space-8) + 1px); */
	position: relative;
	text-align: center;
	transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
	border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
	vertical-align: middle;

	&__loader{
		left: 50%;
		position: absolute;
		top: 50%;
		-webkit-transform: translateX(-50%) translateY(-50%);
		-moz-transform: translateX(-50%) translateY(-50%);
	}

	&__icon {
		align-items: center;
		color: $color-base-gray-20;
		display: flex;
		height: 100%;
		justify-content: center;
		left: 0;
		position: absolute;
		top: 0;
		transition: border-color 0.15s ease-in-out;
		width: 2rem;
	}

	&.Button--icon {
		padding: 0;
		svg {
			margin : auto!important;
		}
	}

	&.Button--icon-alignment-right{
		svg:first-child{
		float: right;
		margin-left: 0.625rem;
		margin-right: 0;
		}
	}

	&:focus {
		box-shadow: 0px 0px 0px 4px #BBB4F4;
		outline: none;
	}

	&--state {
		&-disabled {
			background-color: $color-brand-secondary-lighter !important;
			border: 1px solid transparent;
			color: $color-font-tertiary !important;
			opacity: $opacity-5;
			pointer-events: none;
			text-shadow: none !important;
			& svg path {
				fill: $color-font-tertiary !important
			}
		}
		&-loading{
			cursor: default;
			pointer-events: none;
		}
	}

	&--style {
		&-circle {
			border-radius: $radii-50;
			padding: 0;
		}

		&-fullwidth {
			width: 100%;
		}
	}

	&--primary {
		background-color: $color-brand-primary-base;
		color: $color-font-white;
		text-shadow: 0 1px 0 rgba(0, 0, 0, 0.16);

		&:hover {
			background-color: $color-brand-primary-darker;
		}

		&:focus {
			box-shadow: 0px 0px 0px 4px #BBB4F4;
		}

		&:active {
			background-color: $color-brand-primary-dark;
		}

		svg:first-child {
			margin: 0 $space-10 0 0;
			vertical-align: top;
		}
	}

	&--secondary {
		background-color: $color-brand-white-base;
		border: 1px solid $color-border-link;
		color: $color-font-link;
		
		&:hover{
			border:1px solid $color-base-purple-25;
			color:$color-base-purple-25;
		}

		svg:first-child {
			margin: 0 $space-10 0 0;
			vertical-align: top;
		}

		&:hover {
			border : 1px solid $color-base-purple-25;
			color: $color-brand-primary-darker;
		}

		&:focus {
			box-shadow: 0px 0px 0px 4px #BBB4F4;
		}

		&:active {
			border: 1px solid $color-brand-primary-dark;
			color: $color-brand-primary-dark;
		}
	}

	&--tertiary {
		color: $color-font-link;
		font-size: $size-font-large;
		font-weight: $font-weight-regular;
		
		svg:first-child {
			margin: 0 $space-10 0 0;
			vertical-align: top;
			path {
				fill: $color-font-link;
			}
		}
	}

	&--tertiary-outline {
		background: $color-brand-white-base;
		border: .0625rem solid $color-base-gray-5;
		color: $color-font-base;
		font-size: $size-font-large;
		font-weight: $font-weight-regular;
		height: 2.5rem;
		padding: .4375rem .625rem;

		&:hover {
			border: .0625rem solid $color-border-link;
			color: $color-font-link;

			svg:first-child {
				path, circle, rect {
					fill: $color-font-link;
					stroke: $color-font-link;
				}
			}
		}
		
		svg:first-child {
			margin: 0 $space-10 0 0;
			vertical-align: top;
		} 
	}

	&--outline {
		background: $color-brand-white-base;
		border: 1px solid $color-border-dark;
		color: $color-font-base;
		font-size: $size-font-large;
		font-weight: $font-weight-regular;

		&:hover {
			border: 1px solid $color-border-link;
			color: $color-font-link;
		}
	}

	&--success {
		background-color: $color-brand-white-base;
		color: $color-font-success;
		&:hover {
			background-color: $color-brand-success-base;
			border: 1px solid $color-base-white-5;
			color: $color-font-white;
		}
	}

	&--control {
		border: 1px solid $color-base-gray-5;
		border-radius: $radii-8;
		color: $color-base-gray-20;
		font-size: $size-font-large;
		font-weight: $font-weight-regular;
		line-height: 1;
		max-width: 11.563rem;
		overflow: hidden;
		padding: $space-12 $space-12 $space-12 $space-30;
		position: relative;
		text-overflow: ellipsis;
		white-space: nowrap;

		&:hover {
			border-color: $color-brand-primary-base;
			color: $color-brand-primary-base;
			
			svg {
				path {
				stroke: $color-brand-primary-base;
				}
			}
		}
	}

	&--delete {
		border: 1px solid $color-border-warning;
		color: $color-font-warning;
		svg {
			margin: 0 $space-10 0 0;
			vertical-align: top;
			path {
				fill : $color-base-orange-5;
			}
		}
		&:hover {
			background-color: $color-brand-warning-base;
			color: $color-font-white;
			svg {
				path {
					fill : $color-base-white-5;
				}
			}
		}
	}

	&--outline-delete {
		background-color: $color-brand-white-base;
		border: 1px solid $color-border-link;
		color: $color-font-link;
		svg {
			margin: 0 .625rem 0 0;
			path {
				fill : $color-base-purple-5;
			}
		}
		&:hover{
			background-color:$color-brand-warning-base;
			border: 1px solid $color-border-warning;
			color: $color-font-white;
			svg {
				path {
					fill : $color-base-white-5;
				}
			}
		}
	}

	&--outline-success {
		background-color: $color-brand-white-base;
		border: 1px solid $color-border-success;
		color: $color-font-success;
		svg {
			margin: 0 .625rem 0 0;
			vertical-align: bottom;
		}
		&:hover {
			background-color: $color-brand-success-base;
			color: $color-font-white;
			svg {
				path {
					fill : $color-base-white-5;
					stroke: $color-base-white-5;
				}
			}
		}
	}
	
	&--bulk-delete{
		background-color: $color-brand-primary-base;
		color: $color-font-white;
		svg {
			margin: 0 .625rem 0 0;
			path {
				fill : $color-font-white;
			}
		}
		&:hover{
			background-color: $color-font-white;
			color: $color-base-orange-5;
			svg {
				path {
					fill : $color-base-orange-5;
				}
			}
		}
	}

	&--danger {
		background-color: $color-brand-white-base;
		color: $color-font-warning;
		&:hover {
			background-color: $color-brand-warning-base;
			border: 0.0625rem solid $color-base-white-5;
			color: $color-font-white;
		}
	}

	&--warning {
		background-color: $color-brand-white-base;
		color: $color-font-attention;
		&:hover {
			background-color: $color-brand-attention-base;
			border: 0.0625rem solid $color-base-white-5;
			color: $color-font-white;
		}

		&-link {
			color: $color-base-purple-5;
			font-weight: $font-weight-regular;
			text-decoration: none;
		}
	}

	&--light {
		color: $color-brand-secondary-base;
		&:hover {
			background-color: $color-brand-secondary-lighter;
			color: $color-font-base;
		}
	}

	&--white {
		background: $color-brand-white-base;
		box-shadow: 0 8px 8px rgba(0, 0, 0, 0.02), 0 4px 8px rgba(0, 0, 0, 0.06), 0 8px 16px rgba(0, 0, 0, 0.04);
		color: $color-font-secondary;
		font-weight: $font-weight-regular;
		svg {
			margin-right: $space-10;
		}
		&:hover {
			background: $color-brand-secondary-lighter;
			color: $color-font-black;
			svg path {
				fill : $color-base-gray-20;
			}
		}
	}

	&--size {
		&-large {
			font-size: $size-font-xl;
			padding: $space-12 $space-18 $space-12;
		}

		&-small {
			font-size: $size-font-medium;
			min-height: $space-25;
			padding: $space-5 $space-10;
		}

		&-fullwidth {
			width: 100%;
		}
	}

	&:active {
		border: 1px solid $color-brand-primary-dark;
	}

	.badge {
		margin-left: $space-10;

		&:first-child {
			margin: 0;
			margin-right: $space-10;
		}
	}
}

[type='button']:not(:disabled),
[type='reset']:not(:disabled),
[type='submit']:not(:disabled),
button:not(:disabled) {
  	cursor: pointer;
}

.Button-wrapper button {
  	margin: 0 calc(var(--space-2) + 1px);
}

.Button-container {
	height: 6.25rem;
	margin: $space-15;
	padding: $space-34 0;
	text-align: center;
	width: 12.5rem;
}
