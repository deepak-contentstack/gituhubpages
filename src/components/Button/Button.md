# Button

## Best practices

- Use clear and accurate labels.
- Start button labels with verbs, for example "Add field", not "New field".
- Keep the position of the buttons consistent on the interface.
- Reduce the complexity by using a small number of actions on the interface. It may create confusion if too many actions are used.
