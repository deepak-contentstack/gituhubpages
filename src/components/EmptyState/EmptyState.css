@import '../../../tokens/variables.scss';
@import '../../../tokens/tokens.css';
@import '../../static/styles/css/typography.css';
@import '../../static/styles/css/common.css';

.EmptyStateWrapper {
    font-family: $font-family-primary;
    width: 100%;

    &--card {
        height: calc(100vh - 9.375rem);
    }
    &--list {
        height: calc(100vh - 12.25rem);
    }
    &--fullPage {
        left: 50%;
        position: absolute;
        top: 50%;
        transform: translate(-50%, -50%);
    }
    &--empty{
        height: auto !important;
    }

    .EmptyState {
        max-width: 25rem;
        position: static;
        top: 0;
        transform: none;

        &__description {
            color: $color-brand-secondary-base;
            font-size: $size-font-medium;
            font-weight: $font-weight-medium;
            line-height: $space-16;
            margin-top: $space-20;
        }

        &__footer {
            color: $color-brand-secondary-base;
            font-size: $size-font-small;
            font-weight: $font-weight-medium;
            line-height: $space-16;
            margin-top: $space-20;
        }

        &--primary,&--secondary,&--tertiary {
            text-align: center;
            .EmptyState__actions{
                margin-top: $space-20;
            }
        }

        &--primary {
            .EmptyState__title {
                color: $color-brand-secondary-dark;
                font-size: $size-font-3-xl;
                font-weight: $font-weight-extra-bold;
                letter-spacing: -0.00019em;
                line-height: 1.6875rem;
                margin-top: 0;
            }
        }
    
        &--secondary{
            .EmptyState__title{
                color: $color-brand-secondary-dark;
                font-size: $size-font-2-xl;
                font-weight: $font-weight-bold;
                line-height: $space-22;
            }
        }

        &__title {
            &.EmptyState__title--large {
                color: $color-font-secondary;
                font-size: $size-font-3-xl;
                font-weight: $font-weight-extra-bold;
            }
            &.EmptyState__title--small {
                color: $color-brand-secondary-light;
                font-size: $size-font-medium;
                font-weight: $font-weight-bold;
                line-height: $space-20;
            }
        }
        
        .EmptyState__link {
            color: $color-font-link;
            cursor: pointer;
            font-size: $size-font-medium;
            font-weight: $font-weight-medium;
            line-height: $space-16;
            text-decoration: none;
            svg{
                margin-right: $space-5;
                path{
                    stroke: $color-font-link;
                }
            }
        }
    
        .Button--primary{
            svg{
                path{
                    stroke: $color-brand-white-base;
                }
            }
        }
    }
}

