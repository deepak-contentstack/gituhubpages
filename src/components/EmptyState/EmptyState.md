## **Default**

There are two types of EmptyState components that you can use :

- **Primary** : This is the default type. Use it when you want to display a bolder title.
- **Secondary** : Use this type when you want to display a less bolder information.


<!-- # EmptyState Component

## Props for EmptyState component:
- heading :-
    - This is an optional prop.
    - It is of type `string` or `ReactNode` i.e you can either send the title of the page as a string or as a component.
    - This is a prop that is used to set the title of the page.
- description :- 
   - This is an optional prop.
   - It is of type `string` or `ReactNode` i.e you can either send the description of the page as a string or as a component.
   - This is a prop that is used to set the description of the page. 
- img :- 
    - This is an optional prop.
    - This is of type `ReactNode`.
    - This is mainly to display the image if you want to display on the top of the page.
    - You can send a custom image using this prop.
- imgSrc :-
    - Another way to display image by using this prop where you need to send the src and the alt of the image.
    - This is an optional prop.
    - It is of type `object` which takes two more props further.
        - src :-
            - This is a required prop if you are using imgSrc.
            - It is of type `string` where you need to send the src of the image you want to display.
        - alt :- 
            - This is an optional prop.
            - It is of type `string`.
            - If you want to the `<img>` tag to have an `alt`, you can use this prop.
- displayImage :- 
    - This is an optional props.
    - It is of type `boolean`
    - If you don't want to display the default image if not passed by the user, then use this prop and set the prop to `false`.
- children :- 
    - This is optional.
    - If you want to render any children or you want to render a custom component which consist of header, description and image, you can use this.
    - You need to simply wrap the component you want to display with `<EmptyState></EmptyState>`.
    - It is of type `ReactNode`

## Note :- 
- If you are opting for sending the image, then use either `img` or `imgSrc`.  -->