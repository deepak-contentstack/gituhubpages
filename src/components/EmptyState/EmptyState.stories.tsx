import React from 'react';
import EmptyState from './EmptyState';
import DocSubData from '../../utils/components/DocSubData';
import Button from '../Button/Button';
import Icon from '../Icon2/Icon';
import notes from './EmptyState.md';

const componentDescription = "The EmptyState component lets you display a message to the user if an element cannot be displayed and states the reason why."
const usageDescription = "Empty state can be used when there is no data to show. It helps the user to provide an explanation or guidance to progress."

const emptyStateTypeArray: any = ['primary', 'secondary']
const headingTypeArray: any = ['large', 'small']
const moduleIconArray: any = ['NotificationModuleLarge', 'DiscussionModuleLarge', 'FolderModuleLarge', 'CTModuleLarge', 'TasksModuleLarge', 'PublishQueueLarge', 'ReleaseModuleLarge', 'GFModuleLarge']

export default {
    title: 'Components /EmptyState',
    component: EmptyState,
    argTypes: {
        type: {
            control: { type: 'select', options: emptyStateTypeArray },
            description: 'Lets you specify what type of EmptyState component you want to use. By default, the type is set to “primary.”',
            table: {
                type: {
                    summary: 'string'
                },
                defaultValue: {
                    summary: 'primary'
                }
            }
        },
        className: {
            control: 'disabled',
            description: 'Provide the class names to be appended to this prop.',
            table: {
                type: {
                    summary: 'string',
                },
                defaultValue: {
                },
            },
        },
        moduleIcon: {
            control: { type: 'select', options: moduleIconArray },
            description: 'Lets you pass the name of the icon you want to display.',
            table: {
                type: {
                    summary: 'string'
                },
                defaultValue: {
                },
            }
        },
        heading: {
            control: { type: 'text' },
            description: 'Lets you set the title of a page.',
            table: {
                type: {
                    summary: 'ReactReactNode | string'
                },
                defaultValue: {
                },
            }
        },
        headingType: {
            control: { type: 'select', options: headingTypeArray },
            description: 'Describes the style of the heading of the component.',
            table: {
                type: {
                    summary: 'string'
                },
                defaultValue: {
                },
            }
        },
        description: {
            control: { type: 'text' },
            description: 'Lets you set and display the description of the page as a string or as a component.',
            table: {
                type: {
                    summary: 'ReactReactNode | string'
                },
                defaultValue: {
                },
            }
        },
        actions: {
            control: 'disabled',
            description: 'Lets you pass actions that can be passed to the EmptyState and are displayed below the description.',
            table: {
                type: {
                    summary: 'ReactReactNode'
                },
                defaultValue: {
                },
            }
        },
        footer: {
            control: 'disabled',
            description: 'Lets you pass a footer that can be passed to the EmptyState and is displayed below the actions.',
            table: {
                type: {
                    summary: 'ReactReactNode'
                },
                defaultValue: {
                },
            }
        },
        displayImage: {
            control: 'disabled',
            description: 'Gives you the provision to display or not your image. To hide your display image, set the prop to “false”.',
            table: {
                type: {
                    summary: 'boolean'
                },
                defaultValue: {
                    summary: true
                },
            }
        },
        img: {
            control: 'disabled',
            description: 'Lets you use a custom image.',
            table: {
                type: {
                    summary: 'ReactReactNode | undefined'
                },
                defaultValue: {
                },
            }
        },
        imgSrc: {
            control: 'disabled',
            description: 'Lets you send the src and alt values for an image.',
            table: {
                type: {
                    summary: 'object | undefined'
                },
                defaultValue: {
                },
            }
        },
        children: {
            control: 'disabled',
            description: 'Lets you pass a custom component to render as a child component inside the empty state.',
            table: {
                type: {
                    summary: 'ReactReactNode'
                },
                defaultValue: {
                    summary: null
                }
            }
        },
        width: {
            control: { type: 'text' },
            description: 'Sets the width for the emptyState.',
            table: {
                type: {
                    summary: 'string | Number',
                },
                defaultValue: {
                },
            },
        },
        testId: {
            control: 'disabled',
            description: 'An ID used for testing purposes applied as a data attribute (data-test-id).',
            table: {
                type: {
                    summary: 'string'
                },
                defaultValue: {
                    summary: 'cs-empty-state'
                },
            }
        },

        "Following are the prop values that you can pass under imgSrc:": {
            control: 'disabled',
            description: '',
            table: {
                type: {
                },
                defaultValue: {
                },
            },
        },
        alt: {
            control: 'disabled',
            description: 'Lets you pass an `alt` attribute to your `<img>` tag.',
            table: {
                type: {
                    summary: 'string'
                },
                defaultValue: {
                },
            }
        },
        src: {
            control: 'disabled',
            description: 'Lets you pass the `src` of the image you want to display in the `<img>` tag.',
            table: {
                type: {
                    summary: 'string'
                },
                defaultValue: {
                },
            }
        },
    },
    parameters: {
        design: {
            type: 'figma',
            url: 'https://www.figma.com/proto/drvF57vp3VTJyiCkBSdrbk/Empty-States-%26-CTA-Loader--CS---Venus?node-id=84%3A6810&scaling=min-zoom&page-id=84%3A0',
        },
        componentSubtitle: <DocSubData componentDescription={componentDescription} usageDescription={usageDescription} />,
    },
};

const tableDisable = { table: { disable: true } }

export const Default = (args) => {
    const Footer = () => {
        return <Button onClick={() => { alert('clicked') }} buttonType="tertiary" > Learn More</Button >
    }
    const ChildComponent = () => {
        return <p className='mt-10' > This is the child component</p >
    }
    return <EmptyState
        type={args.type}
        heading={args.heading}
        headingType={args.headingType}
        moduleIcon={args.moduleIcon}
        description={args.description}
        actions={<><div><Button onClick={() => { alert('clicked') }} buttonType="secondary" icon="UnpublishAsset" className="mr-20">Unpublish</Button><Button onClick={() => { alert('clicked') }} buttonType="primary" icon="CheckedWhite">Add</Button></div></>}
        footer={<Footer />}
        testId='custom-emptystate'
        width={args.width}
        children={<ChildComponent />}
    // imgSrc={{ src: 'https://picsum.photos/536/354', alt: 'Error code' }}
    // displayImage={false}
    // img={<Icon icon="LoginError" />}
    />
}

Default.args = {
    type: 'primary',
    heading: 'Structure your content with Content Types!',
    description: 'Think of content type as collection of fields or a template that can be used to create similar kinds of content. You can create different content types for different elements of your project (e.g., homepage, header, navigation)',
    moduleIcon: 'WorkflowModuleLarge',
}

Default.parameters = {
    docs: {
        storyDescription: notes
    }
}

export const CustomComponent = (args) => {
    const MockHeading = () => {
        return <>No matching results found!</>
    };
    const MockDescription = () => {
        return <><div>Try changing the search query or filter to find what you are looking for.</div><Button className="mt-10" buttonType="outline">Learn More</Button></>
    };
    return <EmptyState heading={<MockHeading />} description={<MockDescription />} />
};

CustomComponent.argTypes = {
    type: tableDisable,
    heading: tableDisable,
    headingType: tableDisable,
    description: tableDisable,
    moduleIcon: tableDisable,
    img: tableDisable,
    imgSrc: tableDisable,
    src: tableDisable,
    alt: tableDisable,
    actions: tableDisable,
    footer: tableDisable,
    children: tableDisable,
    width: tableDisable,
    testId: tableDisable,
    className: tableDisable,
    displayImage: tableDisable,
    "Following are the prop values that you can pass under ImgSrcProps:": tableDisable
}

CustomComponent.parameters = {
    docs: {
        storyDescription: `Lets you add a component with a custom-sized heading and description.`
    }
}
