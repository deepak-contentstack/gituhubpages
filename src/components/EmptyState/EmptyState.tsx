import React from 'react';
import cn from 'classnames';
// import Icon from '../Icon2/Icon';
import './EmptyState.css';
import withDeprecatedProp from '../../utils/hooks/depricatedPropsHoc';


type imgSrcProps = {
  /**
  * Lets you pass an “alt” attribute to your <img> tag.
  */
  alt?: string
  /**
  * Lets you pass the src of the image you want to display in the <img> tag.
  */
  src: string
};

export type emptyStateProps = {
  /**
  * Lets you specify what type of EmptyState component you want to use. By default, the type is set to “primary.”
  */
  type?: 'primary' | 'secondary' | 'tertiary'
  /**
  * Provide the class names to be appended to this prop.
  */
  className?: string
  /**
  * Lets you pass the name of the icon you want to display.
  */
  moduleIcon?: string
  /**
  * Lets you set the title of a page.
  */
  heading?: string | React.ReactNode
  /**
  * Describes the style of the heading of the component.
  */
  headingType?: 'large' | 'small',
  /**
  * Lets you set and display the description of the page as a string or as a component.
  */
  description?: string | React.ReactNode
  /**
  * Lets you pass actions that can be passed to the EmptyState and are displayed below the description.
  */
  actions?: React.ReactNode
  /**
  * Lets you pass a footer that can be passed to the EmptyState and is displayed below the actions.
  */
  footer?: React.ReactNode
  /**
  * Gives you the provision to display or not your image. To hide your display image, set the prop to “false”.
  */
  displayImage?: boolean
  /**
  * Lets you use a custom image.
  */
  img?: React.ReactNode
  /**
  * Lets you send the src and alt values for an image.
  */
  imgSrc?: imgSrcProps
  /**
  * Lets you pass a custom component to render as a child component inside the empty state.
  */
  children?: React.ReactNode
  /**
  * Lets you determine for which page you need to use the emptyState.
  */
  forPage?: 'card' | 'list' | 'fullPage' | 'empty'
  /**
  * Sets the width for the emptyState.
  */
  width?: number | string
  /**
  * An ID used for testing purposes applied as a data attribute (data-test-id).
  */
  testId?: string
};

type customImageProps = {
  /**
  * Lets you use a custom image.
  */
  img: React.ReactNode | undefined
  /**
  * Lets you send the src and alt values for an image.
  */
  imgSrc: imgSrcProps | undefined
};

const CustomImage: React.FunctionComponent<customImageProps> = props => {
  const { img, imgSrc } = props;
  const src = imgSrc !== undefined ? imgSrc.src : undefined;
  const alt = imgSrc !== undefined ? imgSrc.alt : undefined;

  return (
    img !== undefined ? <>{img}</> : <img src={src} alt={alt} className="Error-image" />
  );
};

const EmptyState = (props: emptyStateProps) => {
  const {
    heading,
    description,
    img,
    imgSrc,
    className,
    type,
    moduleIcon,
    forPage,
    actions,
    headingType,
    footer,
    width,
    testId
  } = props;

  const displayImage = props.displayImage;
  const children = props.children;
  const classNames = cn('EmptyState', className, [`EmptyState--${type}`]);
  const wrapperClassNames = cn('EmptyStateWrapper flex-center', forPage ? `EmptyStateWrapper--${forPage}` : '')
  const customStyles: any = {};

  if (width) {
    customStyles.maxWidth = width;
  }

  return (
    <div className={wrapperClassNames} data-test-id={testId}>
      <div className={classNames} style={customStyles}>
        {displayImage === true ?
          img !== undefined || imgSrc !== undefined ? <div className="EmptyState__image"><CustomImage img={img} imgSrc={imgSrc} /> </div> : null
          : null
        }
        {/* {moduleIcon !== undefined ? <Icon icon={moduleIcon} /> : null} */}
        {heading !== undefined ? <div className={`EmptyState__title ${headingType ? 'EmptyState__title--' + headingType : ''}`}>{props.heading}</div> : null}
        {description !== undefined ? <div className="EmptyState__description">{props.description}</div> : null}
        {actions !== undefined ? <div className="EmptyState__actions">{props.actions}</div> : null}
        {footer !== undefined ? <div className="EmptyState__footer">{props.footer}</div> : null}
        {children}
      </div >
    </div>
  );
};

EmptyState.defaultProps = {
  displayImage: true,
  children: null,
  type: 'primary',
  testId: 'cs-empty-state'
} as Partial<emptyStateProps>;

export default withDeprecatedProp(EmptyState, { 'subType': 'headingType' })