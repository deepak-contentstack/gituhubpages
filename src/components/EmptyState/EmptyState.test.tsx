import { mount } from 'enzyme/build'
import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../Button/Button'
import EmptyState from './EmptyState'

describe('Test cases for EmptyState Component', () => {
    it('renders the default component', () => {
        const output = mount(<EmptyState heading={'Heading'} description={'Error occured'} testId="custom-test-id" />)
        expect(output.find('[data-test-id="custom-test-id"]').length).toBe(1);
    });
    it('renders the Custom Image', () => {
        const output = mount(<EmptyState img={
            <img src='https://picsum.photos/536/354' alt='Error 404' height="250px" width="340px" />}
            heading={'Heading'}
            description={'Error occured'} />);
        expect(output.find('.EmptyState__image').length).toBe(1);
    });
    it('renders the No Image', () => {
        const output = renderer.create(<EmptyState displayImage={false} heading={'No Image'} description={'This error will have no image'} />).toJSON();
        expect(output).toMatchSnapshot();
    });
    it('renders the Component', () => {
        const MockHeading = () => {
            return <span>Mock Heading</span>
        }
        const MockDescription = () => {
            return <span>Mock Description</span>
        }
        const output = renderer.create(<EmptyState heading={<MockHeading />} description={<MockDescription />} />).toJSON();
        expect(output).toMatchSnapshot();
    });
    it('renders the component with width', () => {
        const output = mount(<EmptyState heading={'No Image'} description={'Error occurred'} width='400px' />);
        expect(output.find('.EmptyState').get(0).props.style).toHaveProperty(
            'maxWidth',
            '400px',
        );
    });
    it('renders the component with headingType props', () => {
        const output = mount(<EmptyState heading={'No Image'} description={'Error occurred'} headingType="small" />);
        expect(output.find('.EmptyState__title--small').length).toBe(1);
    });
    it('renders the component with moduleIcon', () => {
        const output = mount(<EmptyState heading={'No Image'} description={'Error occurred'} moduleIcon="WorkflowModuleLarge" />);
        expect(output.find('[moduleIcon="WorkflowModuleLarge"]').debug()).toContain('WorkflowModuleLarge')
    });
    it('renders the component with footer', () => {
        const Footer = () => {
            return <p>Dummy text</p>
        }
        const output = mount(<EmptyState heading={'No Image'} description={'Error occurred'} footer={<Footer />} />);
        expect(output.find('.EmptyState__footer').length).toBe(1);
    });
    it('renders the component with actions', () => {
        const Actions = () => {
            return <Button>Add</Button>
        }
        const output = mount(<EmptyState heading={'No Image'} description={'Error occurred'} actions={<Actions />} />);
        expect(output.find('.EmptyState__actions').length).toBe(1);
    });
    it('renders the component forPage props', () => {
        const output = mount(<EmptyState heading={'No Image'} description={'Error occurred'} forPage="empty" />);
        expect(output.find('[forPage="empty"]').debug()).toContain('empty')
    });
    it('renders the Component using src props', () => {
        const output = mount(<EmptyState imgSrc={{ src: 'https://picsum.photos/536/354' }} heading={'Heading'} description={'Error occured'} />);
        expect(output.find('[src="https://picsum.photos/536/354"]').debug()).toContain
            ('https://picsum.photos/536/354')
    });
    it('renders the Component using alt props', () => {
        const output = mount(<EmptyState imgSrc={{ src: 'https://picsum.photos/536/354', alt: 'alternative text' }} heading={'Heading'} description={'Error occured'} />);
        expect(output.find('[alt="alternative text"]').debug()).toContain('alternative text')
    });
})
