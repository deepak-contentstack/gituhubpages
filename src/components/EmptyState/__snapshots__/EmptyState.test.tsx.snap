// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for EmptyState Component renders the Component 1`] = `
<div
  className="EmptyStateWrapper flex-center"
  data-test-id="cs-empty-state"
>
  <div
    className="EmptyState EmptyState--primary"
    style={Object {}}
  >
    <div
      className="EmptyState__title "
    >
      <span>
        Mock Heading
      </span>
    </div>
    <div
      className="EmptyState__description"
    >
      <span>
        Mock Description
      </span>
    </div>
  </div>
</div>
`;

exports[`Test cases for EmptyState Component renders the No Image 1`] = `
<div
  className="EmptyStateWrapper flex-center"
  data-test-id="cs-empty-state"
>
  <div
    className="EmptyState EmptyState--primary"
    style={Object {}}
  >
    <div
      className="EmptyState__title "
    >
      No Image
    </div>
    <div
      className="EmptyState__description"
    >
      This error will have no image
    </div>
  </div>
</div>
`;
